In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [ ]:
df= pd.read_csv('BuyerRatio.csv')

In [ ]:
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
# Make a contingency table
df_table=df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
# Chi2 contengency independence test
stat, p, dof, expected=chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, degree of freedom, expected obsvations)

In [ ]:
stat

1.595945538661058

In [ ]:
p

0.6603094907091882

In [ ]:
dof

3

Compare p_value with α = 0.05

Assume Null Hypothesis as Ho: Independence of categorical variables (male-female buyer rations are similar across regions)
(does not vary and are not related)

Thus Alternate Hypothesis as Ha: Dependence of categorical variables (male-female buyer rations are NOT similar across regions
(does vary and somewhat/significantly related)

Inference: As (p-value = 0.6603) > (α = 0.05); Accept the Null Hypothesis 
i.e. Independence of categorical variables Thus, male-female buyer rations are similar across regions and are not related


# Another method

In [ ]:
#to get Degree of freedom
no_of_rows=len(df_table.iloc[0:2,0])
no_of_columns=len(df_table.iloc[0,0:4])
degree_of_freedom=(no_of_rows-1)*(no_of_columns-1)
print('Degree of Freedom=',degree_of_freedom)


Degree of Freedom= 3


In [ ]:
print('dof=%d' % dof)
print(expected)

dof=3
[[  42.76531299  146.81287862  131.11756787   72.30424052]
 [ 442.23468701 1518.18712138 1355.88243213  747.69575948]]


In [ ]:
#to get probability value,critical value,and chi-statistic value 
alpha = 0.05
prob=1-alpha
critical = chi2.ppf(prob, dof)
print('probability=%.3f, critical=%.3f, stat=%.3f' % (prob, critical, stat))
if abs(stat) >= critical:
	print('Dependent (reject H0)')
else:
	print('Independent (fail to reject H0)')

probability=0.950, critical=7.815, stat=1.596
Independent (fail to reject H0)


In [ ]:
# comparing p-value with alpha
print('significance=%.3f, p=%.3f' % (alpha, p))
if p <= alpha:
	print('Dependent (reject H0)')
else:
	print('Independent (fail to reject H0)')

significance=0.050, p=0.660
Independent (fail to reject H0)
